tools/autograd/gen_variable_type.py

```diff
@@ -64,7 +64,7 @@
 baseType->${method_prefix_derived}${base_name}(${unpacked_args})""")
 
 UNPACK_TENSOR = CodeTemplate("""\
-auto${ref} ${arg_name}_ = unpack${suffix}(${arg_name}, "${arg_name}", ${arg_pos});""")
+auto${ref} ${arg_name}_ = unpack${suffix}(${arg_name}, "${arg_name}", ${arg_pos} ${,extra_args});""")
```

Comment:
instead of making this generic can you just name the argument, i.e. scalar_type or opt_scalar_type